dask/dataframe/shuffle.py

```diff
@@ -85,16 +85,28 @@ def sort_values(
     """See DataFrame.sort_values for docstring"""
     if na_position not in ("first", "last"):
         raise ValueError("na_position must be either 'first' or 'last'")
-    if not isinstance(by, str):
-        # support ["a"] as input
-        if isinstance(by, list) and len(by) == 1 and isinstance(by[0], str):
-            by = by[0]
-        else:
-            raise NotImplementedError(
-                "Dataframe only supports sorting by a single column which must "
-                "be passed as a string or a list of a single string.\n"
-                "You passed %s" % str(by)
-            )
+    if not isinstance(by, list):
+        by = [by]
+    if len(by) > 1 and df.npartitions > 1:
+        raise NotImplementedError(
```

Comment:
I think it would still be helpful to do a `str` check here. Naively, I tried passing in a `dd.Series` for `by` (pandas accepts this, and different parts of the dask API accept this as well, such as `set_index`). On `main` this leads to the helpful `NotImplementedError`, but with this change the error is much more opaque because it gets further than this validation step.

Now, it should also be possible to write this function so it takes a `dd.Series`, but that's probably outside the scope of this PR.

Reply:
Ended up consolidating this all into the same check, with a error message that provides a general overview of the expected input and when multi-column sorting is available